parser_library/src/workspaces/workspace.cpp

```diff
@@ -65,6 +67,36 @@ bool workspace::program_id_match(const std::string& filename, const program_id&
     return std::regex_match(filename, prg_regex);
 }
 
+void workspace::delete_diags(processor_file_ptr file)
+{
+    file->diags().clear();
+
+    for (const std::string& fname : file->dependencies())
+    {
+        auto dep_file = file_manager_.find_processor_file(fname);
+        if (dep_file)
+            dep_file->diags().clear();
+    }
+
+    auto notified_found = diag_suppress_notified_.emplace(file->get_file_name(), false);
+    if (!notified_found.first->second)
+        show_message("Diagnostics suppressed from " + file->get_file_name() + ", because there is no configuration.");
+    notified_found.first->second = true;
+}
+
+void workspace::show_message(const std::string& message)
+{
+    if (message_consumer_)
+        message_consumer_->show_message(message, message_type::MT_INFO);
+}
+
+int64_t workspace::get_suppress_diags_limit()
+{
+    if (suppress_diags_limit_)
+        return suppress_diags_limit_.value();
+    return lib_config::get_instance()->diag_supress_limit;
```

Comment:
shouldn't the lib_config instance be just another member of the workspace?

Reply:
The configuration can be changed by the user during lifetime of workspace, so we need to get the instance every time.

Reply:
I was thinking that perhaps each workspace might have different limits?

Reply:
Yes, right now the static lib_config instance represents the configuration of VS Code client. At the same time, you can override the limit of each workspace by writing the configuration into pgm_conf.json.

Reply:
I can move the static instance to workspace_manager, would that be preferable?
But we still need some code to merge the global and workspace configurations.

Reply:
I think that you should not have a static instance at all.

Reply:
Sorry for late editing of the previous comment.

I meant adding the instance as a non-static member of workspace manager.